src/api/procedures/__tests__/setIssuancesData.ts

```diff
@@ -0,0 +1,196 @@
+    entityMockUtils.initMocks();
+    stringToTickerStub = sinon.stub(utilsModule, 'stringToTicker');
+    ticker = 'someTicker';
+    rawTicker = polkadotMockUtils.createMockTicker(ticker);
+  });
+
+  beforeEach(() => {
+    mockContext = polkadotMockUtils.getContextInstance();
+    stringToTickerStub.withArgs(ticker, mockContext).returns(rawTicker);
+    addTransactionStub = procedureMockUtils.getAddTransactionStub();
+  });
+
+  afterEach(() => {
+    entityMockUtils.reset();
+    procedureMockUtils.reset();
+    polkadotMockUtils.reset();
+  });
+
+  afterAll(() => {
+    entityMockUtils.cleanup();
+    procedureMockUtils.cleanup();
+    polkadotMockUtils.cleanup();
+  });
+
+  test('should throw an error if security token is divisible and at least one balance exceeds six decimals', () => {
+    const args = {
+      issuances: [
+        {
+          did: 'someDid',
+          balance: new BigNumber(100),
+        },
+        {
+          did: 'anotherDid',
+          balance: new BigNumber(50.1234567),
+        },
+      ],
+      ticker,
+    };
+
+    entityMockUtils.configureMocks({
+      securityTokenOptions: {
+        details: {
+          isDivisible: true,
+        },
+      },
+    });
+
+    const proc = procedureMockUtils.getInstance<Params, SecurityToken>();
+    proc.context = mockContext;
+
+    return expect(prepareSetIssuancesData.call(proc, args)).rejects.toThrow(
+      'At most one balance exceeds the six decimals limit'
+    );
+  });
+
+  test('should throw an error if security token is not divisible and at least one balance has decimals', () => {
+    const args = {
+      issuances: [
+        {
+          did: 'someDid',
+          balance: new BigNumber(100),
+        },
+        {
+          did: 'anotherDid',
+          balance: new BigNumber(50.1),
+        },
+      ],
+      ticker,
+    };
+
+    const proc = procedureMockUtils.getInstance<Params, SecurityToken>();
+    proc.context = mockContext;
+
+    return expect(prepareSetIssuancesData.call(proc, args)).rejects.toThrow(
+      'At most one balance has decimals'
+    );
+  });
+
+  test('should throw an error if token supply is bigger than the limit total supply', () => {
+    const args = {
+      issuances: [
+        {
+          did: 'someDid',
+          balance: new BigNumber(100),
+        },
+      ],
+      ticker,
+    };
+
+    const limitTotalSupply = new BigNumber(Math.pow(10, 12));
+
+    entityMockUtils.configureMocks({
+      securityTokenOptions: {
+        details: {
+          totalSupply: limitTotalSupply,
+        },
+      },
+    });
+
+    const proc = procedureMockUtils.getInstance<Params, SecurityToken>();
+    proc.context = mockContext;
+
+    return expect(prepareSetIssuancesData.call(proc, args)).rejects.toThrow(
+      `The total supply for "${ticker}" cannot be bigger than ${limitTotalSupply.toString()}`
+    );
+  });
+
+  test('should add a batch issue transaction to the queue', async () => {
```

Comment:
For this test, you should define the balances and dids in arrays, generate the args with map and then use a single forEach to generate the mocks and set up the stubs